packages/web/src/common/clients/wallet-client/transaction-messages/common.ts

```diff
@@ -54,20 +62,50 @@ export function makeTransactionMessage({
   };
 }
 
-export function makeApproveMessage(
-  packagePath: string,
-  args: string[],
+export function makeTokenApproveMessage(
+  tokenPath: string,
+  targetAddress: string,
+  amount: string | bigint | number,
   caller: string,
-) {
+): TransactionMessage {
   return makeTransactionMessage({
     caller,
     send: "",
-    packagePath,
+    packagePath: tokenPath,
     func: "Approve",
-    args,
+    args: [targetAddress, amount.toString()],
   });
 }
 
+export function makeNFTApproveMessage(
+  targetAddress: string,
+  lpTokenId: string | bigint | number,
+  caller: string,
+): TransactionMessage {
+  return makeTransactionMessage({
+    caller,
+    send: "",
+    packagePath: PACKAGE_NFT_PATH,
+    func: "Approve",
+    args: [targetAddress, lpTokenId.toString()],
+  });
+}
+
+export function makeTransactionMessagesWithApproves(
```

Comment:
It would be nice to have validation and error logging and handling.

For example,
```typescript
if (!Array.isArray(transactionMessages) || !Array.isArray(approveInfos)) {
    throw new Error('Invalid input: transactionMessages and approveInfos must be arrays');
  }
```

```typescript
const approveMessages: TransactionMessage[] = [];
const approveResetMessages: TransactionMessage[] = [];

for (const approveInfo of approveInfos) {
  try {
    if (!approveInfo.tokenPath || !approveInfo.targetAddress || approveInfo.amount === undefined || !approveInfo.caller) {
      throw new Error(`Invalid approve info: ${JSON.stringify(approveInfo)}`);
    }
    
    const approveMessage = makeTokenApproveMessage(...);
    approveMessages.push(approveMessage);
    
    const resetMessage = makeTokenApproveMessage(...);
    approveResetMessages.push(resetMessage);
  } catch (error) {
    console.error(`Error creating approve message for: ${JSON.stringify(approveInfo)}`, error);
  }
}
```

What do you think?

Reply:
Thank you for the great suggestion.
It looks better to remove the try catch statement here and propagate the error.

I'll add a custom error.

Reply:
I changed the process of creating the approve message to throw a custom error.
https://github.com/gnoswap-labs/gnoswap-interface/pull/554/commits/8e329fa0c99cc84dbd71ee639cf037e9a47f9c7b#diff-a9ad438b90866ec79f7373a1753edcfb7fef79ce922b8dd47953a8f43da7fac2R102-R124

This allows for clear validation, but I think you should be aware of exception handling when using service functions.

You can handle this particular error with a comparison of `TransactionMessageError`. 
```javascript
try {
    makeTransactionMessagesWithApproves(transactionMessages, approveMessages);
} catch(err) {
    if (err instanceOf TransactionMessageError) {
        console.error("this is transaction message errors");
    }
}
```

Reply:
LGTM! 👍